bin/synapse_stage.py

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env python3
+"""
+Download a Synapse Folder and its contents recursively, or a single Synapse File.
+"""
+import os
+import sys
+
+import synapseclient
+from synapseclient import Folder, File
+
+# Function to recursively download files from a folder
+def download_files_recursively(syn: synapseclient.Synapse, entity_id: str, download_path: str):
```

Comment:
With 4.1 you are able to use
http://python-docs.synapse.org/reference/oop/models_async/#synapseclient.models.Project.sync_from_synapse_async

It's available for the Folder as well. This at least makes it so you won't need to create your own.

Reply:
The latest container might not be created yet. So we should be able to follow this to manually create it
https://sagebionetworks.jira.com/wiki/spaces/SYNPY/pages/643498030?focusedCommentId=3087925256

I can do this later on if you'd like

Reply:
https://hub.docker.com/r/sagebionetworks/synapsepythonclient/tags the newer tags are there

Reply:
Thanks @thomasyu888 , is this currently automated for pushing to hub.docker? I didn't realize this other location existed

Yes, we should include this as a part of our build pipeline though to GHCR